GLMakie/src/GLAbstraction/GLTexture.jl

```diff
@@ -340,6 +340,15 @@ function unsafe_copy!(dest::Array{T, N}, source::Texture{T, N}) where {T,N}
     nothing
 end
 
+# After updating to OpenGL 4.5 or higher, we can use glGetTextureSubImage to copy 
+# a piece of texture back from GPU buffer.
+function unsafe_copy!(dest::Array{T, N}, source::Texture{T, N}, xrange::UnitRange, yrange::UnitRange) where {T,N}
+    bind(source)
+    glReadPixels(first(xrange), first(yrange), length(xrange), length(yrange), source.format, source.pixeltype, dest)
```

Comment:
I think this is only working by chance for you, since the `glReadPixels` works on the currently active framebuffer, which the colorbuffer is likely only by chance.
I'm not 100% sure how to solve this correctly... I think we need to overload this for `FrameBuffer` instead of texture, and then specify the attachment we want to bind...

Reply:
Thanks for the feedback!
So you mean that framebuffer in general does not always contain a `:color` layer with pixel values?
Following your advice I used the `screen.framebuffer.buffers[:color]` (it's in the example code in the draft). Do you think adding this inside the function would not be enough then?
If that is the case, then I would appreciate a hint into what direction to search, because I am still very new to the whole openGL framework.